### ISSUE TITLE
Data tables won't apply fixed header 

### ISSUE THREAD

**Issue by cfarquharson** [wet-boew/wet-boew#7593]
Datatables has a built in ability to fix the header row, per https://datatables.net/extensions/fixedheader/examples/options/enable-disable.html

but when I apply this to the WET4 table, the normal sorting/filter effect shows up but the fixed header part doesn't take affect. `<table class="wb-tables table" data-wb-tables='{"fixedHeader": true}' > 

Is WET4 somehow overriding this?

**Comment by RobJohnston** [comment]
This was @neoinsight 's second point in https://github.com/wet-boew/wet-boew-styleguide/issues/274#issuecomment-227769751.

> The table problem we've seen often, that is far more lethal to task success, is that the table headers don't float. So people scroll down, and then can't map what they're looking at to the header, scroll back up to figure it out, scroll back down to look at the data - argh. But that's a whole separate issue that I should post.

**Comment by neoinsight** [comment]
Yes, apparently, there are lots of things overridden in WET's implementations of the data tables. @steve-h has been able to override some parameters directly in the code on the page (to test out moving the filter field to the left for example), but not all of them can be accessed that way.  

Am disappointed to hear that fixing the header row is being overridden.

**Comment by pjackson28** [comment]
I should caution that the solution in question has accessibility issues. That floating header is actually a separate duplicate table without any data rows and is positioned at the end of the page. It is just a work around to give the visual appearance of the table header floating but structurally it has no association to the table data. So it is similar to using heading markup at the end of a page and using CSS to position it earlier in the page to make it look like the heading for a section of content (without there being any structural relationship between the heading and the section).

**Comment by neoinsight** [comment]
@pjackson28 Thanks - that's too bad that the datatables solution is more of a hack than a solution. Tables are challenging enough for screenreader users, don't want to make them worse!

**Comment by cfarquharson** [comment]
@pjackson28 Can you put aria-hidden="true" on the 2nd table that is just the header?  The visually impaired user will already hear the table header for each cell, it's the sighted user who benefits from the fixed header row and by putting aria-hidden="true" on that floating header, then screen readers will ignore it...right?

**Comment by pjackson28** [comment]
@cfarquharson Problem is there are some partially sighted users that use screen readers as well so they will try to have the floating header they can partially see read off to no avail. Not to mention that aria-hidden="true" doesn't work for all screen readers. There is also the issue that it would still break WCAG 2.0 AA (since structural table heading markup is being used for visual and not structural purposes). On top of that is that the floating heading is still in the tabbing order at the end of the page if the table is still visible (without the rest of the table being in the tabbing order). So numerous issues with the approach.

It is too bad that the browsers don't provide a proper floating table header solution that is tied into the accessibility API.

Has anyone tried dynamically applying the following for a table header when the table header has scrolled partially off the top of the screen (and the rest of the table is still visible)?
```css
thead.float-header {
     position: fixed;
     top: 0;
     background: #FFF;
}
```
Don't know if it would work cross-browser (including on mobile and on screen readers) but if it did then it should address the concerns that I raised above (since the structure would remain intact).

**Comment by pjackson28** [comment]
Note that I used the above CSS when playing around in the IE11 DOM explorer which seemed to cause the table header to stick to the top of the page when I scrolled (although there were still some display issues needing CSS tweaking). So a possible solution that the WET devs could look into that may or may not be possible to turn into a workable solution.

**Comment by hsrudnicki** [comment]
Can be this closed or otherwise? Please update the status of this again, thx

**Comment by RitchieD** [comment]
This is still an open issue.  WET should support fixed headers on datatables.  Our clients also requested this.

**Comment by gdanova** [comment]
We also have clients requesting it. I'd suggest keeping it open unless there is an accessibility show-stopper.

**Comment by Christopher-O** [comment]
> @cfarquharson Problem is there are some partially sighted users that use screen readers as well so they will try to have the floating header they can partially see read off to no avail. Not to mention that aria-hidden="true" doesn't work for all screen readers. There is also the issue that it would still break WCAG 2.0 AA (since structural table heading markup is being used for visual and not structural purposes). On top of that is that the floating heading is still in the tabbing order at the end of the page if the table is still visible (without the rest of the table being in the tabbing order). So numerous issues with the approach.
> 
> It is too bad that the browsers don't provide a proper floating table header solution that is tied into the accessibility API.
> 
> Has anyone tried dynamically applying the following for a table header when the table header has scrolled partially off the top of the screen (and the rest of the table is still visible)?
> 
> ```css
> thead.float-header {
>      position: fixed;
>      top: 0;
>      background: #FFF;
> }
> ```
> 
> Don't know if it would work cross-browser (including on mobile and on screen readers) but if it did then it should address the concerns that I raised above (since the structure would remain intact).

Since we don't support IE11 anymore (as far as I know) could we replace position: fixed, with position: sticky?
That's supported on all current browsers.   
The downside is that on it's own it doesn't work with table-responsive out of the box due to the overflow style on it, though this post shows there is a javascript solution for it
https://uxdesign.cc/position-stuck-96c9f55d9526
